Messaging.md

```diff
@@ -53,13 +57,43 @@ Implementations MUST use a single connection per peer; contract messages (which
 
 ## Message Format
 
-We reuse the [Lightning Message Format](https://github.com/lightningnetwork/lightning-rfc/blob/master/01-messaging.md#lightning-message-format) and the [Type-Length-Value Format](https://github.com/lightningnetwork/lightning-rfc/blob/master/01-messaging.md#type-length-value-format) (TLV).
-To be clear, any encoded binary blob that can be sent over the wire will follow the Lightning Message Format
-while all sub-types internal to these messages will follow the Type-Length-Value Format.
-This means that types on outer-messages will be represented with `u16` integers (defined below) and their length
-is omitted from their encoding because the transport layer has the length in a separate unencrypted field.
-Meanwhile all typed sub-messages (which follow TLV format) will have their types represented using `bigsize` integers
-(defined below) and their lengths (also `bigsize`) are included in their encodings.
+We reuse the [Lightning Message Format](https://github.com/lightningnetwork/lightning-rfc/blob/master/01-messaging.md#lightning-message-format) for messages sent over the wire, the [Type-Length-Value Format](https://github.com/lightningnetwork/lightning-rfc/blob/master/01-messaging.md#type-length-value-format) (TLV) for message extensibility and extra features support, and a custom format for [sub-types](#sub-types).
+
+### Wire Messages
+
+Any encoded binary blob that can be sent over the wire will follow the Lightning Message Format.
+This means that wire messages are prefixed with a `u16` field (defined below) and their length is omitted from their encoding because the transport layer has the length in a separate un-encrypted field..
+
+### Type-Length-Value
+
+A TLV stream is defined for each wire message, and can be used to extend the protocol and for application specific features.
+See the [Type-Length-Value Format section](https://github.com/lightningnetwork/lightning-rfc/blob/master/01-messaging.md#type-length-value-format) in the LN BOLTs for specifications.
+
+### Sub-types
+
+Wire messages (and sub-types themselves) contain embedded data structures.
+These sub-types can have three functions:
+* [Plain sub-types](#plain-sub-type) to factor out a number of fields to make specifications clearer,
+* [Sibling sub-types](#sibling-sub-type) to support multiple variants of a field,
+* [Optional sub-types](#optional-sub-type) to support optional fields.
+
+#### Plain sub-type
+
+Plain sub-types do not have any particular format, and their field can simply be replaced in place where they are used.
+
+#### Sibling sub-type
+
+Sibling sub-types are prefixed with a `bigsize` type identifier.
+The type identifiers are specific to a set of type variants, and can thus be reused across different sub-types.
+Type identifiers are defined within the type definitions, starting from `1` and increasing by `1`.
```

Comment:
My thinking was that the odd/even requirements in LN are used for extensibility, and to determine whether a TLV or message is part of the base protocol or part of a higher layer (application specific) protocol. The way sibling types are defined, there is no case where they should be ignored, so I didn't see a good reason to use the odd/even requirements. If I am missing something and you have in mind a use case for it I'm happy to change.

Reply:
Instead of envisioning sibling type, we could define an even type for a field, servicing as reference and any future, alternative field as an odd one ?

Reply:
@ariard sorry I completely missed your comments. I don't really get what you mean here though, the point of sibling type is not for extensibility or upgradeability but just for cases where within a data structures, a field can have different types.